src/main/scala/is/hail/utils/TextTableReader.scala

```diff
@@ -13,16 +13,19 @@ import scala.util.matching.Regex
 
 case class TableReaderOptions(
   nPartitions: Int,
-  commentStartsWith: Array[String] = Array.empty[String],
-  commentRegexes: Array[Regex] = Array.empty[Regex],
-  separator: String = "\t",
-  missing: String = "NA",
-  noHeader: Boolean = false,
-  header: String = "",
-  quote: java.lang.Character = null,
-  skipBlankLines: Boolean = false) {
-
-  val isComment = TextTableReader.isCommentLine(commentStartsWith, commentRegexes) _
+  commentStartsWith: Array[String],
+  commentRegexes: Array[Regex],
+  separator: String,
+  missing: String,
+  noHeader: Boolean,
+  header: String,
+  quote: java.lang.Character,
+  skipBlankLines: Boolean,
+  useColIndices: Array[Int],
+  originalType: TStruct) {
```

Comment:
`originalType` is only being used through `originalType.size`. Is there any reason to have the whole struct in `TableReaderOptions`?

Reply:
I think it may be possible / nicer to refactor this to use the subtyping machinery in my pruning branch, in which case only the original type and new type are needed (the useColIndices can be computed from these).